### DISCUSSION TITLE
Why does Tailwind use the outdated CommonJS? Time to switch to native ES modules?

### DISCUSSION THREAD

**Discussion by martinszeltins** [tailwindlabs/tailwindcss#10407] [Ideas]
I don't understand why Tailwind uses the old and outdated CommonJS module system when the frontend community has moved to the newer ES modules system?

Maybe it's time to switch to native ES modules?

For example:

````
module.exports = {
  content: ["./src/**/*.{html,js}"],
}
````

**Comment by adamwathan** [comment]
It was released in 2017 and migrating is a big project with lots of challenges, and the ROI just isn't there compared to investing in other areas of the ecosystem. Definitely something we'd like to do eventually but other higher priority stuff to tackle first.

**Reply by martinszeltins** [reply]
Thanks @adamwathan for the answer, that is understandable. Maybe if someone is willing to help out some day it can be migrated.
